src/Shared/Recruit.Vacancies.Client/Infrastructure/Services/Geocode/GeocodeServiceFactory.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Esfa.Recruit.Vacancies.Client.Domain.Repositories;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Esfa.Recruit.Vacancies.Client.Infrastructure.Services.Geocode
+{
+    public class GeocodeServiceFactory : IGeocodeServiceFactory
+    {
+        private readonly GeocodeConfiguration _config;
+        private readonly IVacancyRepository _repository;
+        private readonly ILogger<GeocodeService> _logger;
+
+        public GeocodeServiceFactory(IOptions<GeocodeConfiguration> config, ILogger<GeocodeService> logger, IVacancyRepository repository)
+        {
+            _config = config.Value;
+            _repository = repository;
+            _logger = logger;
+        }
+
+        public IGeocodeService GetGeocodeService()
+        {
+            //1) Try to resolve using the geocode from an existing vacancy with the same postcode

```

Comment:
I think the classes say what they will do.

Reply:
👍